Update Terraform data security object names (#570)
hashicorp/terraform-docs-common: website/docs/cloud-docs/architectural-details/data-security.mdx

| Object                               | Storage      | Encrypted                |
| :----------------------------------- | :----------- | :----------------------- |
| Terraform Configuration (VCS data)   | Blob Storage | Vault Transit Encryption |

> | Private Registry modules             | Blob Storage | Vault Transit Encryption |

Reviewer: ```suggestion
| Private Registry Modules             | Blob Storage | Vault Transit Encryption |
```

I think we actually need to update all of the tables to match our general style guide, but just updating this line to match the style used in the rest of doc.

| Terraform Plan Result                | Blob Storage | Vault Transit Encryption |
| Terraform Audit Trails               | PostgreSQL   | No                       |
| Organization/Workspace/Team Settings | PostgreSQL   | No                       |

### Terraform Cloud and Enterprise Secrets 
| Object                               | Storage      | Encrypted                |
| :----------------------------------- | :----------- | :----------------------- |
| Account Password                     | PostgreSQL   | bcrypt                   |
| 2FA Recovery Codes                   | PostgreSQL   | Vault Transit Encryption |
| SSH Keys                             | PostgreSQL   | Vault Transit Encryption |
| User/Team/Organization Tokens        | PostgreSQL   | HMAC SHA512              |
| OAuth Client ID + Secret             | PostgreSQL   | Vault Transit Encryption |
| OAuth User Tokens                    | PostgreSQL   | Vault Transit Encryption |

### Terraform Cloud Specific 

| Object                               | Storage      | Encrypted                |
| :----------------------------------- | :----------- | :----------------------- |

> | Cloud Data Backups                   | Blob Storage | SSE-S3                   |

Reviewer 2: ```suggestion
| Cloud Data Backups                   | Amazon S3 | SSE-S3                   |
```

> | Cloud Copy of Backups for DR         | Blob Storage | SSE-S3                   |

Reviewer 2: ```suggestion
| Cloud Copy of Backups for DR         | Amazon S3 | SSE-S3                   |
```
